platform-validation/scripts/aziot-compatibility.sh

```diff
@@ -576,11 +577,13 @@ check_storage_space() {
         return
     fi
 
-    if ! echo "$application_size" | grep -q '^[0-9]*.[0-9]*$'; then
+    if ! echo "$application_size" | grep -q '^[.0-9]*$'; then
```

Comment:
This new regex still lets empty string pass through without an error.
```bash
$ echo '' | grep -q '^[.0-9]*$'
$ echo $?
0
```

Reply:
I am having a very bad regex day :D, I pushed another change, and tested the empty case as well